### ISSUE TITLE
gcc 11.1.1 without flags + s390x x86/avx512/shldv failures

### ISSUE THREAD

**Issue by junaruga** [simd-everywhere/simde#866]
I faced the following test failures on gcc 11.1.1 without flags on s390x with simde the latest master commit `c4aa8b492aea70a3a09852c4e0427e07791bb49f` on Fedora rawhide (= 35). Here is the result from the log. Could you take a look at it? Thank you.

```
== 1. tests on gcc ==
+ gcc --version
gcc (GCC) 11.1.1 20210623 (Red Hat 11.1.1-6)
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ g++ --version
g++ (GCC) 11.1.1 20210623 (Red Hat 11.1.1-6)
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ bash -
=== 1.1. tests on gcc without flags ===
+ BUILD_DIR=build/gcc
+ CC='gcc -fno-strict-aliasing'
+ CXX='g++ -fno-strict-aliasing'
+ _setup
+ meson setup build/gcc
The Meson build system
Version: 0.58.0
Source dir: /builddir/build/BUILD/simde-c4aa8b492aea70a3a09852c4e0427e07791bb49f
Build dir: /builddir/build/BUILD/simde-c4aa8b492aea70a3a09852c4e0427e07791bb49f/build/gcc
Build type: native build
Project name: SIMDe
Project version: 0.7.3
C compiler for the host machine: gcc -fno-strict-aliasing (gcc 11.1.1 "gcc (GCC) 11.1.1 20210623 (Red Hat 11.1.1-6)")
C linker for the host machine: gcc -fno-strict-aliasing ld.bfd 2.36.1-15
C++ compiler for the host machine: g++ -fno-strict-aliasing (gcc 11.1.1 "g++ (GCC) 11.1.1 20210623 (Red Hat 11.1.1-6)")
C++ linker for the host machine: g++ -fno-strict-aliasing ld.bfd 2.36.1-15
Host machine cpu family: s390x
Host machine cpu: s390x
Library m found: YES
Compiler for C supports arguments -fopenmp-simd: YES 
Compiler for C++ supports arguments -fopenmp-simd: YES 
Compiler for C supports arguments -Wpsabi: YES 
Compiler for C++ supports arguments -Wpsabi: YES 
Library m found: YES
Build targets in project: 1442
Found ninja-1.10.2 at /usr/bin/ninja
+ _build
+ rm -f build.log
+ _time ninja -C build/gcc -v -j 3
+ head -4 build.log
+ /usr/bin/time -f '=> [%E]' ninja -C build/gcc -v -j 3
ninja: Entering directory `build/gcc'
[1/2884] gcc -fno-strict-aliasing -Itest/common/common-c.p -Itest/common -I../../test/common -I. -I../.. -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -g -fopenmp-simd -DSIMDE_CONSTRAINED_COMPILATION -DSIMDE_ENABLE_OPENMP -Wno-psabi -DSIMDE_TEST_BARE -MD -MQ test/common/common-c.p/common.c.o -MF test/common/common-c.p/common.c.o.d -o test/common/common-c.p/common.c.o -c ../../test/common/common.c
[2/2884] gcc -fno-strict-aliasing  -o test/common/common-c test/common/common-c.p/common.c.o -Wl,--as-needed -Wl,--no-undefined -Wl,--start-group -lm -Wl,--end-group
+ tail -3 build.log
[2883/2884] g++ -fno-strict-aliasing -Itest/mips/msa/subv-native-cpp.p -Itest/mips/msa -I../../test/mips/msa -I. -I../.. -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -g -fopenmp-simd -DSIMDE_CONSTRAINED_COMPILATION -DSIMDE_ENABLE_OPENMP -Wno-psabi -DSIMDE_TEST_BARE -MD -MQ test/mips/msa/subv-native-cpp.p/meson-generated_.._subv.cpp.o -MF test/mips/msa/subv-native-cpp.p/meson-generated_.._subv.cpp.o.d -o test/mips/msa/subv-native-cpp.p/meson-generated_.._subv.cpp.o -c test/mips/msa/subv.cpp
[2884/2884] g++ -fno-strict-aliasing  -o test/mips/msa/subv-native-cpp test/mips/msa/subv-native-cpp.p/meson-generated_.._subv.cpp.o -Wl,--as-needed -Wl,--no-undefined -Wl,--start-group -lm -Wl,--end-group
=> [17:36.56]
+ _test
+ _time meson test -C build/gcc -q --no-rebuild --print-errorlogs
+ /usr/bin/time -f '=> [%E]' meson test -C build/gcc -q --no-rebuild --print-errorlogs
▶  303/1442 shldv/mm_shldv_epi32                  FAIL          
 303/1442 x86/avx512/shldv/emul/c                 ERROR           0.03s   exit status 1
>>> MALLOC_PERTURB_=229 /builddir/build/BUILD/simde-c4aa8b492aea70a3a09852c4e0427e07791bb49f/build/gcc/test/x86/avx512/shldv-emul-c
――――――――――――――――――――――――――――――――――――― ✀  ―――――――――――――――――――――――――――――――――――――
stderr:
../../test/x86/avx512/shldv.c:54: assertion failed: r[0] == simde_x_mm_loadu_epi32(test_vec[i].r)[0] (2069233664 == 2069328821)
(test program exited with status code 1)
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
▶  304/1442 shldv/mm_shldv_epi32                  FAIL          
▶  305/1442 shldv/mm_shldv_epi32                  FAIL          
 304/1442 x86/avx512/shldv/native/c               ERROR           0.03s   exit status 1
>>> MALLOC_PERTURB_=170 /builddir/build/BUILD/simde-c4aa8b492aea70a3a09852c4e0427e07791bb49f/build/gcc/test/x86/avx512/shldv-native-c
――――――――――――――――――――――――――――――――――――― ✀  ―――――――――――――――――――――――――――――――――――――
stderr:
../../test/x86/avx512/shldv.c:54: assertion failed: r[0] == simde_x_mm_loadu_epi32(test_vec[i].r)[0] (2069233664 == 2069328821)
(test program exited with status code 1)
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
 305/1442 x86/avx512/shldv/emul/cpp               ERROR           0.03s   exit status 1
>>> MALLOC_PERTURB_=245 /builddir/build/BUILD/simde-c4aa8b492aea70a3a09852c4e0427e07791bb49f/build/gcc/test/x86/avx512/shldv-emul-cpp
――――――――――――――――――――――――――――――――――――― ✀  ―――――――――――――――――――――――――――――――――――――
stderr:
test/x86/avx512/shldv.cpp:54: assertion failed: r[0] == simde_x_mm_loadu_epi32(test_vec[i].r)[0] (2069233664 == 2069328821)
(test program exited with status code 1)
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
▶  306/1442 shldv/mm_shldv_epi32                  FAIL          
 306/1442 x86/avx512/shldv/native/cpp             ERROR           0.03s   exit status 1
>>> MALLOC_PERTURB_=245 /builddir/build/BUILD/simde-c4aa8b492aea70a3a09852c4e0427e07791bb49f/build/gcc/test/x86/avx512/shldv-native-cpp
――――――――――――――――――――――――――――――――――――― ✀  ―――――――――――――――――――――――――――――――――――――
stderr:
test/x86/avx512/shldv.cpp:54: assertion failed: r[0] == simde_x_mm_loadu_epi32(test_vec[i].r)[0] (2069233664 == 2069328821)
(test program exited with status code 1)
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
Summary of Failures:
 303/1442 x86/avx512/shldv/emul/c                 ERROR           0.03s   exit status 1
 304/1442 x86/avx512/shldv/native/c               ERROR           0.03s   exit status 1
 305/1442 x86/avx512/shldv/emul/cpp               ERROR           0.03s   exit status 1
 306/1442 x86/avx512/shldv/native/cpp             ERROR           0.03s   exit status 1
Ok:                 1438
Expected Fail:      0   
Fail:               4   
Unexpected Pass:    0   
Skipped:            0   
Timeout:            0   
Full log written to /builddir/build/BUILD/simde-c4aa8b492aea70a3a09852c4e0427e07791bb49f/build/gcc/meson-logs/testlog.txt
Command exited with non-zero status 4
=> [0:14.89]
```

**Comment by junaruga** [comment]
I see the file: https://github.com/simd-everywhere/simde/blob/master/docker/cross-files/s390x-gcc-11.cross to test on gcc 11 on s390x exists. Do you know why these failures were not checked on CI?

**Comment by junaruga** [comment]
I skipped the failures by the following command in gcc s390x case as a temporary workaround.

```
sed -i "/^simde_avx512_families/,/\]/ s/'shldv',/#\0/" meson.build
```

**Comment by junaruga** [comment]
I confirmed it was fixed using the simde 48edfa906d835525e2061fbf6062b7c326d66840 . Thanks!
